Generate changelog for 2.11.7 (#11543)
tarantool/tarantool: changelogs/2.11.7.md

# 2.11.7

Date: 2025-05-28
Tag: 2.11.7

## Overview

2.x is the old stable release series. Users are encouraged to update to the
latest 3.x release.

This is a bugfix release. It resolves 28 issues since the previous
version.

Please, consider the full list of user-visible changes below.

## Compatibility

Tarantool 2.x and 3.x are compatible in the binary data layout, client-server

> protocol, and replication protocol. It means that the updating may be performed

Reviewer: ```suggestion
protocol, and replication protocol. It means that an update may be performed
```

with zero downtime for read requests and the-order-of-network-lag downtime for
write requests.

Please, follow the [upgrade procedure][upgrade] to plan your update actions.

[upgrade]: https://www.tarantool.io/en/doc/latest/book/admin/upgrades/

## Bugs fixed

### Core

* Fixed decoding of escape sequences for single-byte character codes from YAML.
  Before the fix, single-byte character codes between `0x80` and `0xff` would
  be erroneously converted to two-byte UTF-8 code points, for example, `\x80`

>   would be decoded as `\uC280` (gh-8782).

Reviewer 2: Is it about the `yaml` module? I think that it is not closely related to the `core` category. Seems to fit into `lua/yaml` or just `lua` better.

Author: I agree, fixed:
```diff
diff --git a/changelogs/2.11.7.md b/changelogs/2.11.7.md
index 3bbf29206a..0381634b1f 100644
--- a/changelogs/2.11.7.md
+++ b/changelogs/2.11.7.md
@@ -28,10 +28,6 @@ Please, follow the [upgrade procedure][upgrade] to plan your update actions.
 
 ### Core
 
-* Fixed decoding of escape sequences for single-byte character codes from YAML.
-  Before the fix, single-byte character codes between `0x80` and `0xff` would
-  be erroneously converted to two-byte UTF-8 code points, for example, `\x80`
-  would be decoded as `\uC280` (gh-8782).
 * Fixed a bug where passwords are leaked into logs
   when using `listen` and `bootstrap_leader` options (gh-10653).
 * Fixed a bug when cancelling a fiber waiting in WAL queue corrupted the
@@ -74,6 +70,13 @@ Please, follow the [upgrade procedure][upgrade] to plan your update actions.
 
 * Fixed a crash which could happen during relay thread shutdown (gh-9920).
 
+### Lua
+
+* Fixed decoding of escape sequences for single-byte character codes from YAML.
+  Before the fix, single-byte character codes between `0x80` and `0xff` would
+  be erroneously converted to two-byte UTF-8 code points, for example, `\x80`
+  would be decoded as `\uC280` (gh-8782).
+
 ### LuaJIT
 
 Backported patches from the vanilla LuaJIT trunk (gh-11055). The following
```

* Fixed a bug where passwords are leaked into logs
  when using `listen` and `bootstrap_leader` options (gh-10653).
* Fixed a bug when cancelling a fiber waiting in WAL queue corrupted the
  WAL (gh-11078).
* Fixed a bug when the timestamps of snapshots created before the server restart
  were not taken into account with `checkpoint_interval` enabled (gh-9820).
* Fixed a bug when the sizes of `*.xlog` files created before the server restart
  were not taken into account during the `checkpoint_wal_threshold` exceedance
  checks (gh-9811).
* Fixed a bug when transactions in the WAL queue were not rolled back on WAL
  error (gh-11179).

> * Fixed bug when WAL queue was not flushed properly. In particular
>   when building an index of a vinyl space. In the latter case it could lead to
>   the new index missing data from transactions in the queue (gh-11118, gh-11119).

Reviewer: ```suggestion
* Fixed a bug when WAL queue was not flushed properly. In particular
  when building an index of a vinyl space. In the latter case it could lead to
  the new index missing data from transactions in the queue (gh-11118, gh-11119).
```

> * Changed error code of `box.error.INVALID_DEC`. It was mistakenly assigned a
>   value 279. Now it is 277.

Reviewer: ```suggestion
* Changed the error code of `box.error.INVALID_DEC`. It was mistakenly assigned the
  value 279. Now it is 277.
```

> * Fixed a bug where tuples with JSON paths in a sequential
>   field order could not be found in a hash index.

Reviewer: ```suggestion
* Fixed a bug where tuples with JSON paths in the sequential
  field order could not be found in a hash index.
```

### Memtx

* Tarantool allowed to create multikey and functional indexes with
  memtx MVCC enabled, but they were not supported. This led to a crash
  or a panic. Now Tarantool raises an error when one tries to create
  an index of the kind with memtx MVCC enabled (gh-6385, gh-11099).

> * Fixed a bug that caused the memtx MVCC to miss conflicts over key definitions
>   that used the number type or collations (gh-10159, gh-11292).

Reviewer: ```suggestion
* Fixed a bug that caused memtx MVCC to miss conflicts over key definitions
  that used the number type or collations (gh-10159, gh-11292).
```

### Vinyl

* Fixed a bug when the garbage collector purged run files left after a dropped
  space without waiting for compaction completion. The bug could result in
  a compaction failure with a "No such file or directory" error (gh-11163).
* Fixed a bug when `space:truncate()` failed with `ER_TRANSACTION_YIELD`
  (gh-11249).
* Fixed a bug in the tuple cache when a transaction operating in a read view
  could skip a tuple deleted after the read view creation (gh-11294).

### Replication

> * Fixed a crash which could happen during relay thread shut down (gh-9920).

Reviewer: ```suggestion
* Fixed a crash which could happen during relay thread shutdown (gh-9920).
```

### LuaJIT

Backported patches from the vanilla LuaJIT trunk (gh-11055). The following
issues were fixed as part of this activity:

* Fixed closing the report file without samples for `jit.p`.

> * Fixed the OOM error handling during recording of the `__concat` metamethod.

Reviewer: ```suggestion
* Fixed the OOM error handling while recording the `__concat` metamethod.
```

> * Fixed the second `trace.flush()` call for the already flushed trace.

Reviewer: ```suggestion
* Fixed the second `trace.flush()` call for an already flushed trace.
```
